packages/app-builder-lib/src/util/AppFileWalker.ts

```diff
@@ -83,7 +110,21 @@ export class AppFileWalker extends FileCopyHelper implements FileConsumer {
       // but do not filter if we inside node_modules dir
       // update: solution disabled, node module resolver should support such setup
       if (file.endsWith(nodeModulesSystemDependentSuffix)) {
-        return false
+        //  it's a node_modules directory
+
+        //  if includeSubNodeModules is true, then we just do nothing - we
+        //  definitely want it. But if its false (the default), then we match
+        //  and if the match is false, we return false (but if the match is
+        //  true, we do nothing).
```

Comment:
Honestly this comment did not help me understand the code. If you feel the code needs this much explaining then it's better to represent the logic through more variable names or clearer code paths. Comments often get out of date with time.

Reply:
A stylistic choice. I've been on projects that are 50/50 code/comment and code reviews are employed to make sure comments stay up to date. I can easily remove.

Reply:
I'm okay with comments, but I also agree with @b-zurg comment. To me, they read both convoluted to me in what their purpose is other than what an if-statement is already explaining. Happy to help out and contribute to this though.

We'll need some new test cases. Such as for when `includeSubNodeModules` is `true`, `false`, and unspecified in config.

Reply:
I'm currently updating this patch to take into account the feedback from you and @b-zurg. In order to make this 'TypeScript friendly', I'll have to alter the API to this call so let me know if there is a big objection to that. I'll resolve the test failures and add some new test cases as well.